cmip6_downscaling/analysis/analysis.py

```diff
@@ -0,0 +1,183 @@
+    pm.execute_notebook(
+        f'{workdir}/analyses.ipynb', executed_notebook_path, parameters=parameters
+    )  # TODO do these local paths work regardless of execution environment?
+
+    # convert from ipynb to html
+    os.system(f"jupyter nbconvert {executed_notebook_path} --to html")
+
+    blob_service_client = BlobServiceClient.from_connection_string(connection_string)
+    blob_name = web_blob + f'/analyses_{run_id}.html'
+    blob_client = blob_service_client.get_blob_client(container='$web', blob=blob_name)
+    # clean up before writing
+    try:
+        blob_client.delete_blob()
+    except:
+        pass
+
+    #  need to specify html content type so that it will render and not download
+    with open(executed_notebook_html_path, "rb") as data:
+        blob_client.upload_blob(data, content_settings=ContentSettings(content_type='text/html'))
+
+    return executed_notebook_path
+
+
+def load_top_cities(plot=False):
+    cities = pd.read_csv('/home/jovyan/cmip6-downscaling/notebooks/worldcities.csv')
+    top_cities = (
+        cities.sort_values('population', ascending=False)
+        .groupby('country')
+        .first()
+        .sort_values('population', ascending=False)[0:100][['city', 'lat', 'lng']]
+    )
+    additional_cities = [
+        'Seattle',
+        'Los Angeles',
+        'Denver',
+        'Chicago',
+        'Anchorage',
+        'Perth',
+        'Paramaribo',
+        'Fortaleza',
+    ]
+    for additional_city in additional_cities:
+        top_cities = top_cities.append(
+            cities[cities['city'] == additional_city][['city', 'lat', 'lng']]
+        )
+    if plot:
+        ax = plt.axes(projection=ccrs.PlateCarree())
+        ax.stock_img()
+        for (lat, lon) in top_cities[['lat', 'lng']].values:
+            plt.plot(
+                lon,
+                lat,
+                color='blue',
+                marker='o',
+            )
+    return top_cities
+
+
+def select_points(ds, top_cities):
+    cities = ds.sel(
+        lat=xr.DataArray(top_cities.lat.values, dims='cities'),
+        lon=xr.DataArray(top_cities.lng.apply(convert_to_360).values, dims='cities'),
+        method='nearest',  # tolerance=1.5; tolerance doesn't work here! what a shame. not all data will be valid if running a subset
+    )
+    cities = cities.assign_coords({'cities': top_cities.city.values})
+    return cities
+
+
+def grab_top_city_data(ds_list, top_cities):
+    city_ds_list = []
+    for ds in ds_list:
+        city_ds_list.append(select_points(ds, top_cities).compute())
+    return city_ds_list
+
+
+def get_seasonal(ds, aggregator='mean'):
+    if aggregator == 'mean':
+        return ds.groupby('time.season').mean()
+    elif aggregator == 'stdev':
+        return ds.groupby('time.season').std()
+    elif aggregator == 'min':
+        return ds.groupby('time.season').min()
+    elif aggregator == 'max':
+        return ds.groupby('time.season').max()
```

Comment:
I think this could be written more generally: 

```python
return getattr(ds.groupby('time.season'), aggregator)()
```

Perhaps this means you don't need this function after all 🤷

Reply:
great- i'll include it.